### ISSUE TITLE
Pie Chart doesn't show when labels are too long

### ISSUE THREAD

**Issue by Ekliptor** [amcharts/amcharts4#102]
Hi,
I'm using 4.0.0-beta.30 from NPM.

If my labels on the legend are too long, the whole legend gets black and nothing is showing up.
Example: https://ibb.co/ir8pS8
By just truncating the legend labels (no other code changes) the chart looks correct: https://ibb.co/jbUeS8

The code used to plot the chart:
```
    public plotPieChart(chartID: string, labels: string[], data: number[], chartOpts?: PlotChartOptions) {
        let chart = am4core.create(chartID, am4charts.PieChart);
        chart.id = chartID;
        chart.legend = new am4charts.Legend();
        
        // Add data
        chart.data = [];
        if (data.length === 0 || data.length !== labels.length) // convert chart.js legacy data
            return AppF.log("Error plotting pie chart with invalid data", data, labels);

        // iterate over the number of line series
        for (let i = 0; i < data.length && i < labels.length; i++)
        {
            let series = {
                siteName: labels[i],
                count: data[i]
            }
            chart.data.push(series);
        }

        // Add and configure Series
        let pieSeries = chart.series.push(new am4charts.PieSeries());
        pieSeries.dataFields.value = "count";
        pieSeries.dataFields.category = "siteName";
        pieSeries.colors = chartOpts.colorSet;
        if (chartOpts.hideLabels === true) {
            pieSeries.labels.template.disabled = true;
            pieSeries.ticks.template.disabled = true;
        }
        this.chartInstances.push(chart);
    }
```

Called with the following data:
```
<div id="iptvMostCentersChart" class="userChart></div>

var appData = {
   iptvCharts: {
      mostDatacenters:
{"labels":["WorldStream B.V. (WorldStream B.V.)","Cloudflare (Cloudflare)","Datacamp Limited (Datacamp Limited)","Vodafone Spain (Vodafone Spain)","OVH SAS (OVH GmbH)","OVH SAS (OVH SAS)","combahton IT Services (combahton IT Services)","SoftLayer Technologies (SoftLayer Technologies)","TPx Communications (TPx Communications)","Fastly (Fastly)","ServInt (ServInt)","Liquid Web, L.L.C (Liquid Web, L.L.C)","Limelight Networks (Limelight Networks)","RCN (RCN)","Choopa, LLC (Choopa, LLC)","Washington State K-20 Telecommunications Network (Seattle Community College District)","Nobis Technology Group, LLC (Leaseweb USA)","University of California, San Diego (University of California, San Diego)","Global Net Access, LLC (Global Net Access, LLC)","Sonstige"],"data":[641,279,92,52,46,14,11,8,7,6,5,2,2,2,1,1,1,1,1,20]}
}
}
var opts = 
                noDataTxt: "no data",
                hideLabels: true
            };
plotPieChart('iptvMostCentersChart', appData.iptvCharts.mostDatacenters.labels, appData.iptvCharts.mostDatacenters.data, opts);
```

Obviously this is not urgent, as my fix was just to write a function to reduce the label length...

**Comment by martynasma** [comment]
OK, so it looks like it's not caused by label length but rather than legend size.

Once the legend grows so large there's no place for the pie chart itself, all labels and markers grow black.

That's definitely a bug, we're going to be addressing.

Here's my test pen: https://codepen.io/team/amcharts/pen/83d6e12c736b698ca56255dfd5544b79?editors=0010

**Comment by Ekliptor** [comment]
I don't know if a map chart is supposed to support legends, but when I use the world map and add a legend, I also get only 2 black legend boxes - same bug.

Map from here: https://codepen.io/team/amcharts/pen/BrMeoO
```
        let chart = am4core.create(chartID, am4maps.MapChart);
        chart.legend = new am4charts.Legend();
```

**Comment by martynasma** [comment]
A legend on the map does not make a lot of sense. Technically, it tries to display an item for each map series (e.g. MapPolygonSeries). But, i don't think there's a lot of sense to it as well.

The only scenario where legend makes sense on the map is when you add custom items to it. However, it's currently not supported, as we're working on it. Please refer to this issue if you want to keep track of status for legend with custom items: #114

**Comment by martynasma** [comment]
Fixed in [4.0.0-beta.31](https://github.com/amcharts/amcharts4/releases/tag/4.0.0-beta.31).

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
